packages/bpk-component-datatable/src/BpkDataTable.js

```diff
@@ -57,24 +76,34 @@ class BpkDataTable extends Component {
     }
   };
 
-  onHeaderClick = ({ dataKey: sortBy, event }) => {
-    const column = this.props.children.find(
+  onHeaderClick = ({
+    dataKey: sortBy,
+    event,
+  }: {
+    dataKey: string,
+    event: SyntheticEvent<HTMLDivElement>,
+  }) => {
+    const column = React.Children.toArray(this.props.children).find(
       child => child.props.dataKey === sortBy,
     );
 
+    if (!column) {
+      return;
+    }
+
     if (column.props.disableSort === true) {
       return;
     }
 
     // See: https://reactjs.org/docs/events.html#event-pooling
-    const eventTarget = event.target;
+    const eventTarget = event.currentTarget;
```

Comment:
This is an actual code change, it seems like the correct option however

Reply:
What is the thinking behind it being the correct option here?

Reply:
> Note: To get the element instance, like HTMLButtonElement in the example above, it is a common mistake to use event.target instead of event.currentTarget. The reason why you want to use event.currentTarget is that event.target may be the wrong element due to event propagation.

From the Flow docs, https://flow.org/en/docs/react/events/

Reply:
`event.target` isn't typed correctly so it makes things difficult

Reply:
I suppose we could verify that `event.target` is an `Element` rather than `Document` or `Window` which are the two other options, but this change seems like the more correct option anyuway